vpn/tunnel.go

```diff
@@ -129,6 +147,61 @@ func (t *Tunnel) ApplyNetworkSettings(ctx context.Context, ns *NetworkSettingsRe
 	return nil
 }
 
+func (t *Tunnel) Update(update *proto.WorkspaceUpdate) error {
+	msg := &TunnelMessage{
+		Msg: &TunnelMessage_PeerUpdate{
+			PeerUpdate: convertWorkspaceUpdate(update),
+		},
+	}
+	select {
+	case <-t.ctx.Done():
+		return t.ctx.Err()
+	case t.sendCh <- msg:
+	}
+	return nil
+}
+
+func (t *Tunnel) start(req *StartRequest) error {
+	rawURL := req.GetCoderUrl()
+	if rawURL == "" {
+		return xerrors.New("missing coder url")
+	}
+	svrURL, err := url.Parse(rawURL)
+	if err != nil {
+		return xerrors.Errorf("parse url: %w", err)
+	}
+	apiToken := req.GetApiToken()
+	if apiToken == "" {
+		return xerrors.New("missing api token")
+	}
+	var header http.Header
+	for _, h := range req.GetHeaders() {
+		header.Add(h.GetName(), h.GetValue())
+	}
+
+	t.conn, err = t.client.NewConn(
+		t.ctx,
+		svrURL,
+		apiToken,
+		&Options{
+			Headers:           header,
+			Logger:            t.logger,
+			DNSConfigurator:   NewDNSConfigurator(t),
+			Router:            NewRouter(t),
+			TUNFileDescriptor: ptr.Ref(int(req.GetTunnelFileDescriptor())),
```

Comment:
Using the tunnel for logging, DNS config, and router config, as well as having a tun file descriptor are all macOS only.

They could be config switches on `*Tunnel` that we enable in `dylib/lib.go`. In general, it's nice to avoid platform-specific files and branches where possible, since they makes testing harder. There's no reason we can't test these functions outside macOS.

Reply:
Ah, okay, I thought the Windows daemon was going to use the tunnel to know what DNS settings to apply, but I assume it'll just reuse the peer updates.

What do you mean by config switch? Just a `isDarwin` flag on `NewTunnel`, or an entire options struct?

Reply:
I meant config for: 

a) using the protocol as the logger
b) using the protocol for DNS config
c) using the protocol for router config

> Ah, okay, I thought the Windows daemon was going to use the tunnel to know what DNS settings to apply, but I assume it'll just reuse the peer updates.

DNS config here are things like what server to use for `.coder` names, not the actual hostname: IP bindings. I think you're getting your wires crossed.  We program `tailnet.Conn` with the hostname: IP bindings, which programs the embedded DNS server. Then we use a DNS Configurator to program the host OS to query the embedded server for all `.coder` hostnames.